### ISSUE TITLE
[Core] Multi-Threaded Actors are Un-Killable

### ISSUE THREAD

**Issue by ijrsvt** [ray-project/ray#28086]
### What happened + What you expected to happen

When one thread in a Multi-Threaded actor exits, the other threads continue to run, even after `ray.kill`'ing the actor.

### Versions / Dependencies

Ray 2.0
OS: Linux

### Reproduction script

```
import ray
import sys
import time
ray.init()

@ray.remote(max_concurrency=3)
class MultiThreadedActor:
    def f(self):
        while True:
            print("Running f")
            time.sleep(3)
    def g(self):
        while True:
            print("running g")
            time.sleep(3)
    def z(self):
        sys.exit()
m = MultiThreadedActor.remote()
m.f.remote()
m.g.remote()
ray.get(m.z.remote())
ray.kill(m)
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by jovany-wang** [comment]
So you mean after `ray.kill(m)`, the process of the actor `m` is still alive?

**Comment by jovany-wang** [comment]
Hi @ijrsvt , I did ran your script, but found that it crashed at the line `ray.get(m.z.remote())`. The version I use is ray 2.0.0 .
![image](https://user-images.githubusercontent.com/19473861/188800638-1263ac3d-5d95-40e3-8585-654fa91ef3cc.png)

**Comment by ijrsvt** [comment]
@jovany-wang This needs to be run interactively. Running this as a python script hides the problem because Ray shuts down when the script exits.

**Comment by jovany-wang** [comment]
@ijrsvt I see. You are right. I missed to check if the process is alive. so I believe it's a bug. I'll investigate it later.

**Comment by jovany-wang** [comment]
Due to the `MultiThreadedActor::f()` is an infinite loop, so `default_executor_.Join()` never return.
See L86 from the following screenshot:

![image](https://user-images.githubusercontent.com/19473861/188930654-516ef750-58d1-455f-a53b-c32af9350db2.png)

**Comment by jovany-wang** [comment]
I believe the fix could be simple. We should just add a flag `force_stop` or `stop_immediately` to `Stop()`, to avoid the graceful stop.

**Comment by jovany-wang** [comment]
Do you have bandwidth to create a PR to fix it, otherwise maybe I could finish in next 3 days.

**Comment by ijrsvt** [comment]
Sorry, but I do not currently have the bandwidth to fix this.

**Comment by ijrsvt** [comment]
Yes, in the reproduction, the lines from the first two threads continue to
get printed

**Comment by jovany-wang** [comment]
@ijrsvt , after deep diving into this issue, I think this is an expected behavior.
See the warning message https://github.com/ray-project/ray/blob/master/src/ray/core_worker/transport/concurrency_group_manager.cc#L83-L85

If your actor is running an infinite loop, you should interrupt it before stoping it, which looks like the behavior you should do in multiple threading programming pattern.

alternatively, we could provide a force_kill flag.

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
